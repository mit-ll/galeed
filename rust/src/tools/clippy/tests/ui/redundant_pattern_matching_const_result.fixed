// run-rustfix

#![feature(const_result)]
#![warn(clippy::redundant_pattern_matching)]
#![allow(clippy::match_like_matches_macro, unused)]

// Test that results are linted with the feature enabled.

const fn issue_5697() {
    if Ok::<i32, i32>(42).is_ok() {}

    if Err::<i32, i32>(42).is_err() {}

    while Ok::<i32, i32>(10).is_ok() {}

    while Ok::<i32, i32>(10).is_err() {}

    Ok::<i32, i32>(42).is_ok();

    Err::<i32, i32>(42).is_err();

    // These should not be linted until `const_option` is implemented.
    // See https://github.com/rust-lang/rust/issues/67441

    if let Some(_) = Some(42) {}

    if let None = None::<()> {}

    while let Some(_) = Some(42) {}

    while let None = None::<()> {}

    match Some(42) {
        Some(_) => true,
        None => false,
    };

    match None::<()> {
        Some(_) => false,
        None => true,
    };
}

fn main() {}
