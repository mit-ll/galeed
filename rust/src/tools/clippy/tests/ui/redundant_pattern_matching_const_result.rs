// run-rustfix

#![feature(const_result)]
#![warn(clippy::redundant_pattern_matching)]
#![allow(clippy::match_like_matches_macro, unused)]

// Test that results are linted with the feature enabled.

const fn issue_5697() {
    if let Ok(_) = Ok::<i32, i32>(42) {}

    if let Err(_) = Err::<i32, i32>(42) {}

    while let Ok(_) = Ok::<i32, i32>(10) {}

    while let Err(_) = Ok::<i32, i32>(10) {}

    match Ok::<i32, i32>(42) {
        Ok(_) => true,
        Err(_) => false,
    };

    match Err::<i32, i32>(42) {
        Ok(_) => false,
        Err(_) => true,
    };

    // These should not be linted until `const_option` is implemented.
    // See https://github.com/rust-lang/rust/issues/67441

    if let Some(_) = Some(42) {}

    if let None = None::<()> {}

    while let Some(_) = Some(42) {}

    while let None = None::<()> {}

    match Some(42) {
        Some(_) => true,
        None => false,
    };

    match None::<()> {
        Some(_) => false,
        None => true,
    };
}

fn main() {}
