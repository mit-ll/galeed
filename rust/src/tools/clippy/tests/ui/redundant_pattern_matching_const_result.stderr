error: redundant pattern matching, consider using `is_ok()`
  --> $DIR/redundant_pattern_matching_const_result.rs:10:12
   |
LL |     if let Ok(_) = Ok::<i32, i32>(42) {}
   |     -------^^^^^--------------------- help: try this: `if Ok::<i32, i32>(42).is_ok()`
   |
   = note: `-D clippy::redundant-pattern-matching` implied by `-D warnings`

error: redundant pattern matching, consider using `is_err()`
  --> $DIR/redundant_pattern_matching_const_result.rs:12:12
   |
LL |     if let Err(_) = Err::<i32, i32>(42) {}
   |     -------^^^^^^---------------------- help: try this: `if Err::<i32, i32>(42).is_err()`

error: redundant pattern matching, consider using `is_ok()`
  --> $DIR/redundant_pattern_matching_const_result.rs:14:15
   |
LL |     while let Ok(_) = Ok::<i32, i32>(10) {}
   |     ----------^^^^^--------------------- help: try this: `while Ok::<i32, i32>(10).is_ok()`

error: redundant pattern matching, consider using `is_err()`
  --> $DIR/redundant_pattern_matching_const_result.rs:16:15
   |
LL |     while let Err(_) = Ok::<i32, i32>(10) {}
   |     ----------^^^^^^--------------------- help: try this: `while Ok::<i32, i32>(10).is_err()`

error: redundant pattern matching, consider using `is_ok()`
  --> $DIR/redundant_pattern_matching_const_result.rs:18:5
   |
LL | /     match Ok::<i32, i32>(42) {
LL | |         Ok(_) => true,
LL | |         Err(_) => false,
LL | |     };
   | |_____^ help: try this: `Ok::<i32, i32>(42).is_ok()`

error: redundant pattern matching, consider using `is_err()`
  --> $DIR/redundant_pattern_matching_const_result.rs:23:5
   |
LL | /     match Err::<i32, i32>(42) {
LL | |         Ok(_) => false,
LL | |         Err(_) => true,
LL | |     };
   | |_____^ help: try this: `Err::<i32, i32>(42).is_err()`

error: aborting due to 6 previous errors

